### PR TITLE
Add support for Spacemesh encrypted wallets

### PR DESCRIPTION
**Pull request by DaveAppleton**
Added support for the encrypted wallet format

ref: https://github.com/spacemeshos/SMIPS/issues/17

Reworking of related commands.

### CODE REVIEW

**Review by avive — CHANGES_REQUESTED** [review]
Please import code used from smWallet "github.com/spacemeshos/CLIWallet/smWallet" directly into the repo and remove this dependency.

File: client/backend.go [hunk 1]
```diff
@@ -1,58 +1,246 @@
...
 	"bytes"
-	"path"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+	"time"
 
 	xdr "github.com/davecgh/go-xdr/xdr2"
```
Comment (avive) [inline client/backend.go:13]: This repo is 404 for me - github.com/davecgh/go-xdr/xdr2

**Review by avive — CHANGES_REQUESTED** [review]

**Comment by avive** [comment]
Looking good - only issue left is the smWallet dependancy. Thanks for this nice PR!

**Comment by DaveAppleton** [comment]
> Please import code used from smWallet "github.com/spacemeshos/CLIWallet/smWallet" directly into the repo and remove this dependency.

This is inside this repo. If you check go.mod you will see that the module is github.com/spacemeshos/CLIWallet so this is referencing a sub folder.

**Comment by avive** [comment]
> > Please import code used from smWallet "github.com/spacemeshos/CLIWallet/smWallet" directly into the repo and remove this dependency.
> 
> This is inside this repo. If you check go.mod you will see that the module is github.com/spacemeshos/CLIWallet so this is referencing a sub folder.

ok. me bad.

**Pull request merged**